### PR TITLE
FormatWriter: insert newline after packages #1069

### LINKED ISSUE #1069: Insert blank line between package declaration and imports section
This template is a guideline, not a strict requirement.
- **Version**: 1.3.0
- **Integration**: standalone library
- **Configuration**:
```
newlines.alwaysBeforeTopLevelStatements = true
```

## Steps

I ran scalafmt like this:

```scala
import org.scalafmt.Scalafmt
import org.scalafmt.config.{Newlines, ScalafmtConfig}

object ScalaFormat {

  def main(args: Array[String]): Unit = {
    val source =
      """package com.foo.bar
import collection.JavaConverters._
import com.foo.baz.Grunka
trait Entity {
  val kind: Kind
  val uri: String
}"""

    val scalafmtConfig = ScalafmtConfig(newlines = Newlines(alwaysBeforeTopLevelStatements = true))
    val formattedSource = Scalafmt.format(source, scalafmtConfig).get
    print(formattedSource)
  }
}
```

## Problem

Scalafmt formats code like this:
```scala
package com.foo.bar
import collection.JavaConverters._
import com.foo.baz.Grunka

trait Entity {
  val kind: Kind
  val uri: String
}
```

## Expectation

I would like the formatted output to look like this:
```scala
package com.foo.bar

import collection.JavaConverters._
import com.foo.baz.Grunka

trait Entity {
  val kind: Kind
  val uri: String
}
```

## Notes

It would be nice to be able to insert a blank line between `package` declaration and imports section.

### PR DESCRIPTION
**Pull request by kitbellew**
Fixes #1069

This commit produces no diffs on `scala-repos`.

### CODE REVIEW

**Comment by kitbellew** [comment]
@olafurpg @poslegm @tanishiking PTAL

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala [hunk 1]
```diff
@@ -258,13 +258,31 @@ class FormatWriter(formatOps: FormatOps) {
       case x => x
     }
     initStyle.newlines.alwaysBeforeTopLevelStatements && {
-      topLevelTokens.contains(hash(toks(i).formatToken.right)) && {
-        val (distance, FormatToken(_, nextNonComment, _)) =
-          nextNonCommentWithCount(toks(i).formatToken)
-        isMultiline(actualOwner(nextNonComment).tokens.last, i + distance + 1)
+      val formatToken = toks(i).formatToken
+      def checkPackage: Option[Boolean] = {
```
Comment (olafurpg) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: We should guard this change against `edition = 2019-11`

Comment (kitbellew) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: will add the edition here, and post the diffs from scala-repos later today.

for now, working on #1557, i think the solution is incomplete, but chains are surprisingly difficult.

Comment (olafurpg) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: Chains are very difficult with the current router design. I spent a LOT of time trying to break on plain `a.b` without exploding the search state but failed.

**Comment by kitbellew** [comment]
@olafurpg rebased, added edition, as requested. updated description regarding `scala-repos`, this change had no effect on that codebase.

**Pull request merged**